packages/router/src/history.js

```diff
@@ -13,18 +13,13 @@ const history = createBrowserHistory();
 const originalHistoryPush = history.push;
 const originalHistoryReplace = history.replace;
 
-function buildSearch( params ) {
-	const queryString = buildQueryString( params );
-	return queryString.length > 0 ? '?' + queryString : queryString;
-}
-
 function push( params, state ) {
-	const search = buildSearch( params );
+	const search = buildQueryString( params );
```

Comment:
Looks even better and more readable now 👏

Reply:
This is breaking the query params.
This appends query params without a `?` and page-reload fails for this reason.
`http://local.site/wp-admin/site-editor.phpcanvas=edit`